src/Link/Link.tsx

```diff
@@ -29,6 +29,7 @@ const StyledLink = styled.a<StyledLinkProps>`
   text-decoration: none;
 
   /* You can add one by setting underline={true} */
+  /** @deprecated */
```

Comment:
```suggestion
```

I think the notice could live on `StyledLinkProps` above like:

```tsx
type StyledLinkProps = {
  hoverColor?: string
  muted?: boolean
  /** @deprecated use `inline` to specify the type of link instead */
  underline?: boolean
  // Link inside a text block
  inline?: boolean
} & SxProp
```

Hope that makes sense! This should mean that folks get this kind of error in their IDE of choice:

![An editor showing the deprecation message for the underline prop in a code snippet](https://github.com/primer/react/assets/3901764/64ee101c-4f79-4634-9605-8e58140513dc)